### PR TITLE
[EXC-1140] Changes for readOnlyToken

### PR DESCRIPTION
**Pull request by NidaKarimAli**

### CODE REVIEW

File: src/exchange/sockets.ts [hunk 1]
```diff
@@ -118,7 +129,7 @@ export class Sockets {
   }
 
   // Emitted when any price bin on the oderbook is updated.
-  onOrderBookUpdate = (cb: ({ orderbook }: any) => void) => {
+  onOrderBookUpdate = (cb: ({ symbol, orderbook }: any) => void) => {
```
Comment (malikkulsoom) [inline src/exchange/sockets.ts:132]: Don't we have to update the same thing in Websocket.ts?

File: src/fireflyClient.ts [hunk 2]
```diff
@@ -237,19 +237,25 @@ export class FireflyClient {
...
     if (!addresses.ok) {
       throw Error("Failed to fetch contract addresses");
     }
 
-    this.contractAddresses = addresses.data;
+     this.contractAddresses = addresses.data;
 
     // onboard user if not onboarded
-    if (userOnboarding) {
+    if (userOnboarding && !apiToken) {
```
Comment (malikkulsoom) [inline src/fireflyClient.ts:250]: we can do something like for the sake of ease and simplicity

if(apiToken){
        //do xyz..
    }
    else if(userOnboarding){
        //do onboarding
    }

File: tests/fireflyClient.test.ts [hunk 3]
```diff
@@ -1339,3 +1512,506 @@ describe("FireflyClient", () => {
     });
   });
 });
+
+describe("FireflyClient via ReadOnlyToken", () => {
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:1516]: These tests are all happy cases of get route, can we add few post route tests?

File: tests/fireflyClient.test.ts [hunk 4]
```diff
@@ -1082,6 +1083,178 @@ describe("FireflyClient", () => {
     });
   });
 
+  describe("Sockets", () => {
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:1086]: Sorry might be a stupid question but I'm struggling to find out the difference between these socket tests and the ones written above :D Could you point out?

**Review by malikkulsoom — CHANGES_REQUESTED** [review]
You need to bump the version too

File: tests/fireflyClient.test.ts (hunk 4 above)
Comment (NidaKarimAli) [reply tests/fireflyClient.test.ts]: This was an extra describe, thanks!

File: tests/fireflyClient.test.ts (hunk 3 above)
Comment (NidaKarimAli) [reply tests/fireflyClient.test.ts:1343]: I tried to add post APIs as well upon your suggestion, but those APIs before making an HTTP post call, require more data that we do not have when we initialize readOnly client. ex. walletAddress etc, so the API fails with an error that does not directly related with the feature. I have however, added failure case for post route using readOnlyToken on dapi side to validate the forbidden status it expects.

**Review by malikkulsoom — APPROVED** [review]

**Review by malikkulsoom — APPROVED** [review]

**Pull request merged**